### PR TITLE
En 9637/unit tests validator to delegation

### PR DESCRIPTION
**Pull request by sasurobert**
added a set of unit tests for validator to delegation functionality

### CODE REVIEW

**Review by iulianpascalau — APPROVED** [review]

File: vm/systemSmartContracts/delegation_test.go [hunk 1]
```diff
@@ -4246,3 +4246,455 @@ func TestDelegation_GetDelegationManagementShouldWork(t *testing.T) {
...
+	marshaledData, _ = d.marshalizer.Marshal(validatorData)
+	eei.SetStorageForAddress(d.validatorSCAddr, addr, marshaledData)
+
+	eei.returnMessage = ""
+	_, returnCode = d.getAndVerifyValidatorData(addr)
+	assert.Equal(t, returnCode, vmcommon.UserError)
+	assert.Equal(t, eei.returnMessage, "clean unStaked info before changing validator to delegation contract")
+}
+
+func TestDelegation_initFromValidatorData(t *testing.T) {
```
Comment (SebastianMarian) [inline vm/systemSmartContracts/delegation_test.go:4429]: This is the longest unit test I ever seen :)

File: vm/systemSmartContracts/staking.go [hunk 2]
```diff
@@ -258,36 +258,8 @@ func (s *stakingSC) changeValidatorKey(args *vmcommon.ContractCallInput) vmcommo
...
-
-	s.eei.SetStorage(oldKey, nil)
-	err = s.saveStakingData(newKey, stakedData)
-	if err != nil {
-		s.eei.AddReturnMessage("cannot save staking data: error " + err.Error())
-		return vmcommon.UserError
-	}
-
-	return vmcommon.Ok
+	s.eei.AddReturnMessage("function is deprecated")
```
Comment (SebastianMarian) [inline vm/systemSmartContracts/staking.go:261]: Why not remove also L257-L260 if this function is deprecated to be called?

File: vm/systemSmartContracts/staking_test.go [hunk 3]
```diff
@@ -2067,6 +2067,76 @@ func TestStakingSc_UnStakeNodeWhenMaxNumIsMoreShouldNotStakeFromWaiting(t *testi
...
+	arguments.CallValue.SetUint64(10)
+	retCode = sc.Execute(arguments)
+	assert.Equal(t, vmcommon.UserError, retCode)
+	assert.Equal(t, eei.returnMessage, "callValue must be 0")
+
+	arguments.CallValue.SetUint64(0)
+	eei.returnMessage = ""
+	retCode = sc.Execute(arguments)
+	assert.Equal(t, vmcommon.UserError, retCode)
+	assert.Equal(t, eei.returnMessage, "number of arguments is 2 at minimum")
```
Comment (SebastianMarian) [inline vm/systemSmartContracts/staking_test.go:2117]: This error message was not changed in the productive code with "invalid number of arguments" ?

File: vm/systemSmartContracts/delegation_test.go (hunk 1 above)
Comment (sasurobert) [reply vm/systemSmartContracts/delegation_test.go:4429]: A lot of cases are tested here

File: vm/systemSmartContracts/staking.go (hunk 2 above)
Comment (sasurobert) [reply vm/systemSmartContracts/staking.go:261]: As it exists right now in the mainnet, so it must stay for backward compatibility

File: vm/systemSmartContracts/staking_test.go (hunk 3 above)
Comment (sasurobert) [reply vm/systemSmartContracts/staking_test.go:2117]: for this function I think not.

Comment (Oxsy-SIbiu) [reply vm/systemSmartContracts/staking_test.go:2117]: ok

File: vm/systemSmartContracts/staking.go (hunk 2 above)
Comment (SebastianMarian) [reply vm/systemSmartContracts/staking.go:261]: Ok

File: vm/systemSmartContracts/staking_test.go (hunk 3 above)
Comment (SebastianMarian) [reply vm/systemSmartContracts/staking_test.go:2117]: Ok

**Review by SebastianMarian — APPROVED** [review]

**Pull request merged**